### PR TITLE
Reducing cyclomatic complexity score in agent_capability.go

### PR DESCRIPTION
**Pull request by yunhee-l**
### Summary
Refactoring capabilities function into smaller methods, reducing cyclomatic complexity. 
Note: Go Report Card warns on functions with cyclomatic complexity > 15.

before:
➜ gocyclo agent/app/agent_capability.go 
20 app (*ecsAgent).capabilities agent/app/agent_capability.go:61:1
...
after:
➜ gocyclo agent/app/agent_capability.go
8 app (*ecsAgent).capabilities agent/app/agent_capability.go:61:1
5 app (*ecsAgent).appendTaskIamRoleCapabilities agent/app/agent_capability.go:125:1
4 app (*ecsAgent).appendTaskENICapabilities agent/app/agent_capability.go:164:1
4 app (*ecsAgent).appendTaskCPUMemLimitCapabilities agent/app/agent_capability.go:148:1
4 app (*ecsAgent).appendLoggingDriverCapabilities agent/app/agent_capability.go:108:1
...

### Implementation details
Refactored a single method setting agent capability into smaller per-capability logic methods.

### Testing

- [x] Builds on Linux (`make release`)
- [x] Builds on Windows (`go build -out amazon-ecs-agent.exe ./agent`)
- [x] Unit tests on Linux (`make test`) pass
- [x] Unit tests on Windows (`go test -timeout=25s ./agent/...`) pass
- [x] Integration tests on Linux (`make run-integ-tests`) pass
- [x] Integration tests on Windows (`.\scripts\run-integ-tests.ps1`) pass
- [x] Functional tests on Linux (`make run-functional-tests`) pass
- [x] Functional tests on Windows (`.\scripts\run-functional-tests.ps1`) pass

unit tests on linux passed locally (go test ./agent/...)

New tests cover the changes: N/A

### Description for the changelog
Did not update changelog as this is a simple refactoring, not a new feature or bug fix.

### Licensing

This contribution is under the terms of the Apache 2.0 License: yes

### CODE REVIEW

File: agent/app/agent_capability.go [hunk 1]
```diff
@@ -73,6 +73,39 @@ func (agent *ecsAgent) capabilities() ([]*ecs.Attribute, error) {
...
+	}
+
+	if _, ok := supportedVersions[dockerclient.Version_1_19]; ok {
+		capabilities = appendNameOnlyAttribute(capabilities, capabilityPrefix+"ecr-auth")
+		capabilities = appendNameOnlyAttribute(capabilities, attributePrefix+"execution-role-ecr-pull")
+	}
+
+	capabilities = agent.appendTaskIamRoleCapabilities(capabilities, supportedVersions)
+
+	capabilities, err := agent.appendTaskCPUMemLimitCapabilities(capabilities, supportedVersions)
```
Comment (aaithal) [inline agent/app/agent_capability.go]: Can you please check if a unit test already covers this? If not, can you add one?

Comment (yunhee-l) [reply agent/app/agent_capability.go]: https://github.com/aws/amazon-ecs-agent/blob/dev/agent/app/agent_capability_test.go#L393 seems to be covering the TaskCPUMemLimit capability setting. 

However there was a bug before this fix where AWSLogsExecutionRole override setting was dependent on TaskENIPluginVersionAttribute retrieval (https://github.com/aws/amazon-ecs-agent/blob/dev/agent/app/agent_capability.go#L144), but unit tests weren't there to catch it. Maybe I should update AWSLogsExecutionRole testcase to check for no dependency?

Comment (aaithal) [reply agent/app/agent_capability.go]: Yeah, I think we should have something to prevent such regressions in the future.

**Comment by yunhee-l** [comment]
Updated TestCapabilitiesExecutionRoleAWSLogs testcase to include error from TaskENIPluginVersionAttribute retrieval.

**Pull request merged**